mlir-sycl/doc/rebasing.md

```diff
@@ -0,0 +1,65 @@
+# Rebasing `sycl` Branch
```

Comment:
I think there is no rebasing here, only merging.
Furthermore, it is about the `sycl-mlir` branch and not `sycl`.
So, I have the feeling that the title could be something like _Merging `sycl` branch into `sycl-mlir` branch_.
Also the file name could be `merging_from_upstream.md`.

Reply:
Thanks for the comments!